benchmarks/ab_benchmark.py

```diff
@@ -0,0 +1,272 @@
+                     f"-p 8080:8080 -p 8081:8081 " \
+                     f"-v /tmp/benchmark/conf:/opt/ml/conf -v /tmp/benchmark/logs:/home/model-server/logs" \
+                     f" {enable_gpu} -itd {docker_image} torchserve --start --ts-config /opt/ml/conf/config.properties"
+    execute(docker_run_cmd, wait=True)
+    time.sleep(5)
+
+
+def resolve_local_exec_dependency(input_file):
+    shutil.rmtree('/tmp/model_store/', ignore_errors=True)
+    os.makedirs("/tmp/model_store/", exist_ok=True)
+    resolve_common_exec_dependency(input_file)
+
+
+def resolve_docker_exec_dependency(input_file):
+    resolve_common_exec_dependency(input_file)
+
+
+def resolve_common_exec_dependency(input_file):
+    input = input_file
+    shutil.rmtree("/tmp/benchmark", ignore_errors=True)
+    os.makedirs("/tmp/benchmark/conf", exist_ok=True)
+    os.makedirs("/tmp/benchmark/logs", exist_ok=True)
+    shutil.copy('config.properties', '/tmp/benchmark/conf/')
+    shutil.copyfile(input, '/tmp/benchmark/input')
+
+
+def update_exec_params(input_param):
+    for k, v in input_param.items():
+        if default_ab_params[k] != input_param[k]:
+            execution_params[k] = input_param[k]
+
+
+def report_generate():
+    click.secho("\n\nGenerating Reports", fg='green')
+    generate_csv_output()
+    generate_latency_graph()
+
+
+def generate_csv_output():
+    click.secho(" Generating CSV output...", fg='green')
+    execute(f"grep \"PredictionTime\" {metric_log} |  cut -d\"|\" -f1 | cut -d \":\" -f4 > /tmp/benchmark/predict.txt",
+            wait=True)
+    batched_requests = execution_params['requests'] / execution_params['batch_size']
+    line50 = int(batched_requests / 2)
+    line90 = int(batched_requests * 9 / 10)
+    line99 = int(batched_requests * 99 / 100)
+
+    artifacts = {}
+    artifacts['ts_error'] = execute_return_stdout(
+        """grep "Failed requests:" {0} | awk '{1}'""".format(result_file, "{print $NF}"))
+    artifacts['ts_rps'] = execute_return_stdout(
+        """grep "Requests per second:" {0} | awk '{1}'""".format(result_file, "{print $4}"))
+    artifacts['ts_p50'] = execute_return_stdout("""grep "50\%" {0} | awk '{1}'""".format(result_file, "{print $NF}"))
+    artifacts['ts_p90'] = execute_return_stdout("""grep "90\%" {0} | awk '{1}'""".format(result_file, "{print $NF}"))
+    artifacts['ts_p99'] = execute_return_stdout("""grep "90\%" {0} | awk '{1}'""".format(result_file, "{print $NF}"))
+    artifacts['ts_mean'] = execute_return_stdout(
+        """grep -E "Time per request:.*mean\)" {0} | awk '{1}'""".format(result_file, "{print $4}"))
+    artifacts['ts_error_rate'] = execute_return_stdout(
+        """echo "scale=2; 100*{0}/{1}" | bc | awk '{2}'""".format(int(artifacts['ts_error']),
+                                                                  execution_params['requests'],
+                                                                  "{printf \"%f\", $0}"))
```

Comment:
Can we parse this with RegEX instead of calling grep in a subprocress.

Reply:
Changed  to use of regex.